commander3/src/comm_tod_mod.f90

```diff
@@ -470,6 +496,118 @@ subroutine read_tod(self, detlabels)
        end do
     end do
 
+    ! Initialize mean gain
+    allocate(ns(0:self%ndet))
+    self%gain0 = 0.d0
+    ns         = 0
+    do i = 1, self%nscan
+       do j = 1, self%ndet
+          if (.not. self%scans(i)%d(j)%accept) cycle
+          self%gain0(j) = self%gain0(j) + self%scans(i)%d(j)%gain
+          ns(j)         = ns(j) + 1
+       end do
+    end do
+    call mpi_allreduce(MPI_IN_PLACE, self%gain0, self%ndet+1, &
+         & MPI_DOUBLE_PRECISION, MPI_SUM, self%comm, ierr)
+    call mpi_allreduce(MPI_IN_PLACE, ns,         self%ndet+1, &
+         & MPI_INTEGER,          MPI_SUM, self%comm, ierr)
+    self%gain0(0) = sum(self%gain0)/sum(ns)
+    where (ns > 0)
+       self%gain0 = self%gain0 / ns - self%gain0(0)
+    end where
+
+    do i = 1, self%nscan
+       do j = 1, self%ndet
+          self%scans(i)%d(j)%dgain = self%scans(i)%d(j)%gain - self%gain0(0) - self%gain0(j)
+          self%scans(i)%d(j)%baseline = 0d0
+       end do
+    end do
+
+    ! Precompute trigonometric functions
+    allocate(self%sin2psi(self%npsi), self%cos2psi(self%npsi))
+    allocate(self%psi(self%npsi))
+    do i = 1, self%npsi
+       psi             = (i-0.5)*2.0*pi/real(self%npsi,sp)
+       self%psi(i)     = psi
+       self%sin2psi(i) = sin(2.0*psi)
+       self%cos2psi(i) = cos(2.0*psi)
+    end do
+
+    call mpi_barrier(self%comm, ierr)
+    call wall_time(t2)
+    if (self%myid == 0) write(*,fmt='(a,i4,a,i6,a,f8.1,a)') &
+         & '    Myid = ', self%myid, ' -- nscan = ', self%nscan, &
+         & ', TOD IO time = ', t2-t1, ' sec'
+
+  end subroutine read_tod
+
+  subroutine read_tod_WMAP(self, detlabels)
```

Comment:
Does this really need a separate function? Can't we just add some loops to read_tod?

Reply:
yes, I actually think the function is obsolete as a whole, I think I can delete it